libs/sdk-core/src/moonpay/moonpay_api.rs

```diff
@@ -0,0 +1,84 @@
+use anyhow::Result;
+
+use crate::breez_services::BreezServer;
+use crate::moonpay::moonpay_url_signer::MoonPayUrlSigner;
+use crate::SwapInfo;
+
+#[tonic::async_trait]
+pub trait MoonPayApi: Send + Sync {
+    async fn sign_moon_pay_url(&self, url_data: &dyn MoonPayUrlData) -> Result<String>;
+}
+
+#[tonic::async_trait]
+impl MoonPayApi for BreezServer {
+    async fn sign_moon_pay_url(&self, url_data: &dyn MoonPayUrlData) -> Result<String> {
```

Comment:
Not sure we need the MoonPayUrlData trait. Looks simpler just to pass the two values exlicitly.